### ISSUE TITLE
Sega Megadrive/Genesis - allow choosing gamepad type in bml file or through CLI

### ISSUE THREAD

**Issue by Tartifless** [higan-emu/ares#1644]
**Is your feature request related to a problem? Please describe.**
When using ARES through command line, or through a launcher, or on a setup without keyboard / mouse, it is impossible to switch gamepad type to fighting pad or 6-button pad for megadrive/genesis, as this is a per-session setting only.

**Describe the solution you'd like**
**Describe alternatives you've considered**
3 solutions i could see:
1) Make the pad type setting a settng exposed in the .bml file
2) Make the setting available through a command-line launch argument
3) For Genesis/Megadrive, default to 6-button pas instead of 3-button (no incompatibility and most games are far more playable in 6-button mode, especially fighting games)

**Comment by remutro** [comment]
Suggestion number 3 shouldn't be implemented because there are a lot of Mega Drive games that do not work correctly with the 6-button controller. Users would just think there is something wrong with emulating certain games in that configuration if they were not aware there could be an issue. So in the interest of having the largest compatibility, the default should remain the 3 button controller.

Reference: https://segaretro.org/Six_Button_Control_Pad_(Mega_Drive)

**Comment by Tartifless** [comment]
Yes indeed (not a lot but some), which would be in favor of having this properly in the settings file

**Comment by suphamster** [comment]
How is 6 buttons pad expected to work since when I choose it in settings (for Ultimate Mortal Kombat 3 game that is supposed to support it) it reverts to 3 buttons back after closing settings window, but if I even don't close settings window, switch to fighting pad then in game it's still working as 3 button pad like nothing changed and ignores assigned to gamepad (FD Direwolf 3 model) buttons? Same goes (with 6 pad supported) WWF WrestleMania: The Arcade Game. Using latest nighlty version of ares.

![Image](https://github.com/user-attachments/assets/165e9fc6-82cc-47d1-b0d4-ef3223c7c2eb)

![Image](https://github.com/user-attachments/assets/ca91e691-c620-4c58-a5b9-81d5d3a81b83)

**Comment by LukeUsher** [comment]
> How is 6 buttons pad expected to work since when I choose it in settings (for Ultimate Mortal Kombat 3 game that is supposed to support it) it reverts to 3 buttons back after closing settings window, but if I even don't close settings window, switch to fighting pad then in game it's still working as 3 button pad like nothing changed and ignores assigned to gamepad (FD Direwolf 3 model) buttons? Same goes (with 6 pad supported) WWF WrestleMania: The Arcade Game. Using latest nighlty version of ares.
> 
> ![Image](https://github.com/user-attachments/assets/165e9fc6-82cc-47d1-b0d4-ef3223c7c2eb)
> 
> ![Image](https://github.com/user-attachments/assets/ca91e691-c620-4c58-a5b9-81d5d3a81b83)

This is the menu where you map your controls, it doesn’t change what devices are connected to the system

For that, you want the system menu in the ares main menu (for Mega Drive it will say mega drive), there you can see any ports or slots that can be swapped, eg controller ports or discs for cd based systems

**Comment by suphamster** [comment]
> > How is 6 buttons pad expected to work since when I choose it in settings (for Ultimate Mortal Kombat 3 game that is supposed to support it) it reverts to 3 buttons back after closing settings window, but if I even don't close settings window, switch to fighting pad then in game it's still working as 3 button pad like nothing changed and ignores assigned to gamepad (FD Direwolf 3 model) buttons? Same goes (with 6 pad supported) WWF WrestleMania: The Arcade Game. Using latest nighlty version of ares.
> > ![Image](https://github.com/user-attachments/assets/165e9fc6-82cc-47d1-b0d4-ef3223c7c2eb)
> > ![Image](https://github.com/user-attachments/assets/ca91e691-c620-4c58-a5b9-81d5d3a81b83)
> 
> This is the menu where you map your controls, it doesn’t change what devices are connected to the system
> 
> For that, you want the system menu in the ares main menu (for Mega Drive it will say mega drive), there you can see any ports or slots that can be swapped, eg controller ports or discs for cd based systems

Thanks. it works now! So every time I switch the game I would need to choose fighting pad again?

**Comment by Tartifless** [comment]
> > > How is 6 buttons pad expected to work since when I choose it in settings (for Ultimate Mortal Kombat 3 game that is supposed to support it) it reverts to 3 buttons back after closing settings window, but if I even don't close settings window, switch to fighting pad then in game it's still working as 3 button pad like nothing changed and ignores assigned to gamepad (FD Direwolf 3 model) buttons? Same goes (with 6 pad supported) WWF WrestleMania: The Arcade Game. Using latest nighlty version of ares.
> > > ![Image](https://github.com/user-attachments/assets/165e9fc6-82cc-47d1-b0d4-ef3223c7c2eb)
> > > ![Image](https://github.com/user-attachments/assets/ca91e691-c620-4c58-a5b9-81d5d3a81b83)
> > 
> > 
> > This is the menu where you map your controls, it doesn’t change what devices are connected to the system
> > For that, you want the system menu in the ares main menu (for Mega Drive it will say mega drive), there you can see any ports or slots that can be swapped, eg controller ports or discs for cd based systems
> 
> Thanks. it works now! So every time I switch the game I would need to choose fighting pad again?

Yes, everytime you restart ares, this is the request behind this issue posting: being able to choose type of gamepad through the bml file or through cli.

**Comment by Tartifless** [comment]
Hello, 
Would there be a possibility to enable selection of controller type to be stored in the settings.bml file or to be enabled from command line ?

**Comment by awsms** [comment]
> Hello, Would there be a possibility to enable selection of controller type to be stored in the settings.bml file or to be enabled from command line ?

you can try out #2448, which basically lets you map everything that's currently only doable in the GUI, or once a game has booted.

**Comment by Tartifless** [comment]
Thanks, will try this this week

**Comment by Tartifless** [comment]
Hello @awsms : i tried this version and it's working fine.

**Comment by Tartifless** [comment]
@awsms @remutro : any chance to see #2448 merged ? It would be a nice addition to enable selection dualshock or 6-button megadrive pad when launching a game.

**Comment by xii7** [comment]
Is it possible to make to make the fight pad the default?
